server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -505,6 +518,41 @@ private void handleBadRequest(String uri, RestRequest.Method method, RestChannel
         }
     }
 
+    /**
+     * Attempts to extract auth token and login.
+     *
+     * @returns false if there was an error and the request should not continue being dispatched
+     * */
+    private boolean handleAuthenticateUser(final RestRequest request, final RestChannel channel) {
+        try {
+            final AuthToken token = RestTokenExtractor.extractToken(request);
+            // If no token was found, continue executing the request
+            if (token == null) {
+                // Authentication did not fail so return true. Authorization is handled at the action level.
+                return true;
+            }
+            final Subject currentSubject = identityService.getSubject();
+            currentSubject.login(token);
+            logger.info("Logged in as user " + currentSubject);
+        } catch (final Exception e) {
+            try {
+                final BytesRestResponse bytesRestResponse = BytesRestResponse.createSimpleErrorResponse(
+                    channel,
+                    RestStatus.UNAUTHORIZED,
+                    e.getMessage()
+                );
+                channel.sendResponse(bytesRestResponse);
+            } catch (final Exception _ignored) {
```

Comment:
Nit, not ignored anymore :-)

```suggestion
            } catch (final Exception ex) {
```